hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -991,9 +1001,9 @@ private void updateFunctionalIndexIfPresent(HoodieCommitMetadata commitMetadata,
   private HoodieData<HoodieRecord> getFunctionalIndexUpdates(HoodieCommitMetadata commitMetadata, String indexPartition, String instantTime) throws Exception {
     HoodieFunctionalIndexDefinition indexDefinition = getFunctionalIndexDefinition(indexPartition);
     List<Pair<String, FileSlice>> partitionFileSlicePairs = new ArrayList<>();
-    HoodieTableFileSystemView fsView = HoodieTableMetadataUtil.getFileSystemView(metadataMetaClient);
+    HoodieTableFileSystemView fsView = HoodieTableMetadataUtil.getFileSystemView(dataMetaClient);
     commitMetadata.getPartitionToWriteStats().forEach((dataPartition, value) -> {
-      List<FileSlice> fileSlices = getPartitionLatestFileSlicesIncludingInflight(metadataMetaClient, Option.ofNullable(fsView), dataPartition);
+      List<FileSlice> fileSlices = getPartitionLatestFileSlicesIncludingInflight(dataMetaClient, Option.ofNullable(fsView), dataPartition);
```

Comment:
Is there any case when the log files for this instant got cleaned?

Reply:
Cleaner will only delete the older log files which are already committed at least one commit before the latest version. So, log files for this instant should be present.

Reply:
Yeah, the latest file slice is there but the `instantTime` here may not belongs to the latest file slice?

Reply:
Not following you here. In multi-writer scenario, there is a new latest file slice due to instant t1 but `dataMetaClient` had already been initialized before t1 (say upto instant t0), then index will only be updated upto t0. In this case, `getPartitionLatestFileSlicesIncludingInflight` will only return file slices upto t0. The fsView API will return file slices from a consistent snapshot.

Reply:
> Not following you here. In multi-writer scenario, there is a new latest file slice due to instant t1

How about this client triggers cleaning for t0 with a very radical strategy while we do this loading check.

Reply:
Tracking in HUDI-7581